packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2792,7 +2792,7 @@ export class ContainerRuntime
 		const { message, local, localOpMetadata } = messageWithContext;
 
 		// Intercept to reduce minimum sequence number to the delta manager's minimum sequence number.
-		// Sequence numbers are not guaranteed to follow any sort of order. Re-entrancy is one of those situations
+		// Minimum sequence numbers are not guaranteed to follow any sort of order. Re-entrancy is one of those situations
```

Comment:
This comment is still misleading. It's incorrect to say that min seq# are not guaranteed to follow any sort of order. They do follow an order and are always increasing similar to sequence number.
The reason this is needed is because the runtime could be tracking ops in pending state manager whose min seq# is lower than delta manager's because delta manager knows nothing about these pending ops. Basically, container runtime's knowledge of min seq# can be different from delta managers. So, we need to modify the min seq# before sending these ops to the lower layers for correctness.

Reply:
Also, not sure what you mean by "Re-entrancy is one of those situations". This comment should expand on that. What scenario does re-entrancy create that would lead to updating the min seq # here.

Reply:
This is not guaranteed, one of the op re-entrancy tests breaks when I tried to introduce this invariant that minimum sequence number is always increasing.

Reply:
Update: I threw in an assert, it seemed to not run into any issues. I can add an assert here

Reply:
update, I added an assert to validate the invariant, it seems to hold.

Reply:
Okay, good. Irrespective, the statement min squence numbers are not guaranteed to follow order is incorrect. The server will always stamp them in order. The client may have some scenarios where it changes the order while processing but the invariant still holds. So, even if you don't have the assert, I would remove that generic statement :)

Reply:
@markfields fyi - I know you might be working in this area, so it's best you're informed of this part of the runtime.